### PR TITLE
Cap magnetometer's accepted reading to once per 0.5sec

### PR DESCRIPTION
**Pull request by nirvn**

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for ff078c107042d83db7d799d41f77ccc98f11f274: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3776-49c737e82df4b56519f0953c30140272faf419e5-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3776-49c737e82df4b56519f0953c30140272faf419e5-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3776-49c737e82df4b56519f0953c30140272faf419e5-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3776-49c737e82df4b56519f0953c30140272faf419e5-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3776-49c737e82df4b56519f0953c30140272faf419e5-arm64-android.apk)
- [all access arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3776-49c737e82df4b56519f0953c30140272faf419e5-arm-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3776-49c737e82df4b56519f0953c30140272faf419e5-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3776-49c737e82df4b56519f0953c30140272faf419e5-x86-android.apk)
</details>

File: src/qml/LocationMarker.qml [hunk 1]
```diff
@@ -57,9 +57,14 @@ Item {
         }
     }
 
+    property int lastAcceptedReading: 0
     onReadingChanged: {
-      compassOrientation = userOrientation + (-(Math.atan2(reading.x, reading.y) / Math.PI) * 180)
-      compassHasValue = true
+      var timestamp = Date.now();
+      if (timestamp - lastAcceptedReading > 500) {
```
Comment (nirvn) [inline src/qml/LocationMarker.qml:63]:
```suggestion
      if (timestamp - lastAcceptedReading > 5000) {
```

Comment (suricactus) [reply src/qml/LocationMarker.qml:63]: Isn't this 5 seconds instead of 0.5 secs?

Comment (nirvn) [reply src/qml/LocationMarker.qml:63]: @suricactus , that was just me doing a test, back to 500.

Comment (suricactus) [reply src/qml/LocationMarker.qml:63]: Stupid me, sorry for the noise.

**Review by suricactus — APPROVED** [review]

**Pull request merged**